test/e2e/operator_test.go

```diff
@@ -230,6 +239,64 @@ func TestClusterProxyProtocol(t *testing.T) {
 	}
 }
 
+// TestProxyProtocolAPI verifies that the operator configures router pod
+// replicas to use PROXY protocol if it is specified on an ingresscontroller.
+func TestProxyProtocolAPI(t *testing.T) {
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "proxy-protocol"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newNodePortController(icName, domain)
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, availableConditionsForIngressControllerWithNodePort...); err != nil {
+		t.Errorf("failed to observe expected conditions: %v", err)
+	}
+
+	if err := kclient.Get(context.TODO(), icName, ic); err != nil {
+		t.Fatalf("failed to get ingresscontroller: %v", err)
+	}
+	deployment := &appsv1.Deployment{}
+	if err := kclient.Get(context.TODO(), controller.RouterDeploymentName(ic), deployment); err != nil {
+		t.Fatalf("failed to get ingresscontroller deployment: %v", err)
+	}
+	proxyProtocolEnabled := false
+	for _, v := range deployment.Spec.Template.Spec.Containers[0].Env {
+		if v.Name == "ROUTER_USE_PROXY_PROTOCOL" {
+			if val, err := strconv.ParseBool(v.Value); err == nil {
+				proxyProtocolEnabled = val
+				break
+			}
+		}
+	}
+	if proxyProtocolEnabled {
+		t.Fatalf("expected initial deployment not to enable PROXY protocol")
+	}
+
+	ic.Spec.EndpointPublishingStrategy.NodePort = &operatorv1.NodePortStrategy{
+		Protocol: operatorv1.ProxyProtocol,
+	}
+	if err := kclient.Update(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+	err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
```

Comment:
Could you reuse ` waitForDeploymentEnvVar` here? 

https://github.com/openshift/cluster-ingress-operator/blob/master/test/e2e/operator_test.go#L1864